app/jobs/reservation_job.rb

```diff
@@ -0,0 +1,44 @@
+class ReservationJob < ActiveJob::Base
+  queue_as :default
+
+  def perform
+    if enabled
+      log_start
+      run
+      clean
+      log_completion
+    else
+      log_disabled
+    end
+  end
+
+  private
+
+  def enabled
+    true
+  end
+
+  def log_start(type: '', task: '')
+    Rails.logger.info "Found #{collection.count} #{type} reservations, #{task}"
+  end
+
+  def log_disabled(task: '')
+    Rails.logger.info "Reservations is not configured to #{task}. "\
+      'Please change the application settings if you wish to do so.'
+  end
+
+  def log_completion
+    Rails.logger.info 'Done!'
+  end
+
+  def run
```

Comment:
I was going to recommend raising a `NoMethodError` here by default, although I guess if it does nothing by default then there's no harm in letting it run. It would just be confusing if the logging methods ran with nothing happening. Alternatively, we could have `enabled` default to `false` and then have `log_disabled`'s `task` parameter default to `UNDEFINED TASK` or something so that if we have an improperly set up child task, that its performance will make sense. What do you think?